src/cml.js

```diff
@@ -179,32 +179,32 @@ class CML {
         );
 
       const longReport = `${commitSha.substr(0, 7)}\n\n${report}`;
-      const [commitPr = {}] = await drv.commitPrs({ commitSha });
-      const { url } = commitPr;
-
-      if (!url && !isBB)
-        throw new Error(`PR for commit sha "${inCommitSha}" not found`);
-
-      if (url) {
-        const [prNumber] = url.split('/').slice(-1);
-
-        if (update)
-          comment = updatableComment(await drv.prComments({ prNumber }));
-
-        if (update && comment) {
-          commentUrl = await drv.prCommentUpdate({
-            report: longReport,
-            id: comment.id,
-            prNumber
-          });
-        } else
-          commentUrl = await drv.prCommentCreate({
-            report: longReport,
-            prNumber
-          });
-
-        if (this.driver !== 'bitbucket') return commentUrl;
-      }
+      const commitPrs = await drv.commitPrs({ commitSha });
+      await commitPrs.map((pr) => {
```

Comment:
In any case we need it we would need to await all promises

```js
await Promise.all(commitPrs.map((pr) => {
...
});
```

Reply:
> I have the impression that this might not be needed because of:
> 
> ```js
>  .filter((pr) => pr.state === 'opened')
> ```
> 
> I can confirm it but Im quite sure that you can only have one PR/MR open at once

I agree that we are on some weird corner case where you merge a first PR/MR in two different branches that both have a PR/MR attached to it. Can't say it won't happen though. The question is whether we want to allow this weird flow or not :)

Reply:
what I want to say is because you are filtering the PRs to be opened. We will have just only one every single time not needing to promisify all the PRs returned working as it was.

Reply:
I reverted to previous code ;)